lib/cli.js

```diff
@@ -339,7 +339,14 @@ const cli = {
                 return 0;
             } catch (err) {
                 debug("Error retrieving environment info");
-                log.error(err.message);
+
+                let errorMessage = err.message;
+
+                if (usingFlatConfig) {
+                    errorMessage += "\nYou're using eslint.config.js, some command line flags are no longer available. Please see https://eslint.org/docs/latest/use/command-line-interface for details.";
+                }
+
+                log.error(errorMessage);
                 return 2;
             }
```

Comment:
This will only execute when `--env-info` flag is passed and there's an unexpected error while calculating the data for it.

With `eslint --ext .ts`, I'm still getting the same `Invalid option '--ext' - perhaps you meant '-c'?` as before.

If we want to display this info when `--ext` option is passed in flat config mode, I think we'll need to define `ext` as a valid option for optionator (in `options.js`) in flat config mode and then manually check if it was passed in.

Reply:
> I think we'll need to define `ext` as a valid option for optionator (in `options.js`) in flat config mode and then manually check if it was passed in.

Actually, this might not be a good idea because the option will appear in `--help`, and also because optionator will validate types and can throw an error before we have a chance to display that the option is invalid.

Might be better to parse the error message here:

https://github.com/eslint/eslint/blob/f30cefee6bda2789ede18e1664b84c2638ea1bb5/lib/cli.js#L319-L323

Reply:
Good call. 👍